godot-macros/src/util/mod.rs

```diff
@@ -303,3 +305,50 @@ pub fn venial_parse_meta(
 
     venial::parse_item(input)
 }
+
+// ----------------------------------------------------------------------------------------------------------------------------------------------
+
+// util functions for handling #[func]s and #[var(get=f, set=f)]
+
+pub fn make_function_registered_name_constants(
+    funcs: &[FuncDefinition],
+    class_name: &Ident,
+) -> Vec<TokenStream> {
+    funcs
+        .iter()
+        .map(|func| {
+            make_function_registered_name_constant(
+                class_name,
+                &func.signature_info.method_name,
+                &func.registered_name,
+                &func.external_attributes,
+            )
+        })
+        .collect()
+}
+
+/// funcs can be renamed with `#[func(rename=new_name) fn f();`.
+/// To be able to access the renamed function name at a later point, it is saved in a string constant.
+pub fn make_function_registered_name_constant(
+    class_name: &Ident,
+    func_name: &Ident,
+    exported_name: &Option<String>,
+    attributes: &Vec<Attribute>,
+) -> TokenStream {
+    let const_name = format_function_registered_name_constant(class_name, func_name);
+    let const_value = match &exported_name {
+        Some(renamed) => renamed.to_string(),
+        None => func_name.to_string(),
+    };
+
+    quote! {
+        #(#attributes)*
```

Comment:
at the moment, *all* attributes are copied over. this should probably be changed so only #[cfg] attributes are copied over

Reply:
Maybe `#[cfg_attr]` :thinking:

Reply:
can you elaborate? 

This would be relevant in, for example, func_test.rs
![image](https://github.com/user-attachments/assets/31624c1c-c71e-456a-840a-4711b8c20c84)

Reply:
> Maybe #[cfg_attr]

Oh I think I know what you meant. 

that would only be relevant for the special case ``#[cfg_attr(/**/, cfg)]``, right?